stat/samplemv/halton_test.go

```diff
@@ -0,0 +1,44 @@
+package samplemv
+
+import (
+	"math"
+	"math/rand"
+	"testing"
+
+	"gonum.org/v1/gonum/mat"
+	"gonum.org/v1/gonum/stat/distmv"
+)
+
+func TestHalton(t *testing.T) {
+	for cas, test := range []struct {
+		n int
+		d int
+	}{
+		{10, 1},
+		{100, 2},
+		{1000, 3},
+	} {
+		src := rand.New(rand.NewSource(1))
+		// Generate the samples
+		batch := mat.NewDense(test.n, test.d, nil)
+		Halton(batch, Owen, distmv.NewUnitUniform(test.d, nil), src)
+
+		// In each dimension, the samples should be stratefied according to the
+		// prime for that dimension. There should be at most 1 sample per
+		// 1/b^k block, where k is log(n)/log(b).
+		for d := 0; d < test.d; d++ {
+			b := float64(nthPrime(d))
+			fk := math.Log(float64(test.n)) / math.Log(b)
+			k := math.Ceil(fk)
+			den := math.Pow(b, k)
+			m := make(map[int]struct{})
```

Comment:
Is it worth making this a `map[int]int` and counting them? In the happy case it makes no difference, but when debugging it might help. (I'm being extremely pessimistic here given that this passes).

Reply:
Done.